suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/VisitorStateModifications.java

```diff
@@ -63,60 +75,76 @@ public static Description interceptDescription(VisitorState visitorState, Descri
+
+        // In order to be able to suppress multiple errors in one pass on the same element, we need to do a single
+        // Fix/Replacement in error-prone. It's not possible to do this bit by bit with multiple Replacements. To do
+        // this, we make sure we only make one fix per source element we put the suppression on by using a Map. This
+        // way we have our own mutable Fix that we can add errors to, and only once the file has been visited by all
+        // the error-prone checks it will then produce a replacement with all the checks suppressed.
+        boolean alreadyReportedFix = FIXES.containsKey(firstSuppressibleParent);
+
+        SuppressingFix suppressingFix = FIXES.computeIfAbsent(
+                firstSuppressibleParent,
+                _ignored -> new SuppressingFix(
+                        Optional.ofNullable(visitorState.getSourceCode()), suppressWarnings, firstSuppressibleParent));
+
+        suppressingFix.suppressError(description.checkName);
+
+        // If we already submitted our mutable fix, we don't need to do so again, just need to add the error to the fix.
+        if (alreadyReportedFix) {
+            return Description.NO_MATCH;
+        }
 
         return Description.builder(
                         description.position,
                         description.checkName,
                         description.getLink(),
                         description.getMessageWithoutCheckName())
-                .addFix(SuggestedFix.builder()
-                        .prefixWith(
-                                firstSuppressibleParent,
-                                "@com.palantir.suppressibleerrorprone.RepeatableSuppressWarnings(\""
-                                        + CommonConstants.AUTOMATICALLY_ADDED_PREFIX + description.checkName
-                                        + "\")\n"
-                                        + indent)
-                        .build())
+                .addFix(suppressingFix)
                 .build();
     }
 
-    private static Optional<CharSequence> indentForTree(VisitorState visitorState, Tree firstSuppressibleParent) {
-        return Optional.ofNullable(visitorState.getSourceCode())
-                .map(sourceCode -> whitespaceIndentBefore(
-                        sourceCode, ((DiagnosticPosition) firstSuppressibleParent).getStartPosition()));
+    private static boolean suppressibleTree(Tree tree) {
+        return isModifiersTree(tree).isPresent();
     }
 
-    static CharSequence whitespaceIndentBefore(CharSequence sourceCode, int sourceElementPosition) {
-        int pos = sourceElementPosition - 1;
+    private static Optional<ModifiersTree> isModifiersTree(Tree tree) {
+        // This covers all type definitions eg class, interface, enum, record, annotation, future kinds
+        // of class-like type definitions.
+        if (tree instanceof ClassTree) {
+            return Optional.of(((ClassTree) tree).getModifiers());
+        }
```

Comment:
am I misremembering or was there a reason you couldn't use java 17 here?

Reply:
Some repos internally run the compiler with 11, we had to revert back in #22

Reply:
Looks like 46 repos have been blocked on 6.9.0 because of that change